src/relayFeeCalculator/chain-queries/baseQuery.ts

```diff
@@ -63,11 +59,31 @@ export default abstract class QueryBase implements QueryInterface {
    * Retrieves the current gas costs of performing a fillRelay contract at the referenced Spoke Pool
    * @returns The gas estimate for this function call (multplied with the optional buffer)
    */
-  async getGasCosts(): Promise<BigNumberish> {
-    const tx = await createUnsignedFillRelayTransaction(this.spokePool, this.usdcAddress, this.simulatedRelayerAddress);
+  async getGasCosts(
+    amountToRelay: BN,
```

Comment:
as discussed offline, the interface shoudl require the caller to pass in both `fillAmount` and `amount`. I would rename `amountToRelay` to `fillAmount` to match the smart contracts nomenclature.